UnityProject/Assets/Editor/Tests/DTODeserialisationTests.cs

```diff
@@ -0,0 +1,219 @@
+                    }
+                ]
+            } ";
+
+            ObstaclesDTO obstaclesDTO = JsonUtility.FromJson<ObstaclesDTO>(obstacleJSON);
+            Assert.AreEqual(2, obstaclesDTO.obstacles.Length);
+
+            ObstacleDTO obstacleDTO = obstaclesDTO.obstacles[0];
+            Assert.AreEqual(new Location(0, 1), obstacleDTO.location);
+            Assert.AreEqual(2, obstacleDTO.width);
+            Assert.AreEqual(1, obstacleDTO.height);
+            Assert.AreEqual(ObstacleType.Van, obstacleDTO.ObstacleType);
+            Assert.AreEqual(Orientation.East, obstacleDTO.OrientationType);
+        }
+
+        [Test]
+        public void TestGameUpdateDTODeserialisation()
+        {
+            string gameUpdateJSON = @" {
+                ""era"": ""less_flat"",
+                ""southWestCorner"": {
+                    ""x"": -2,
+                    ""y"": -2
+                },
+                ""northEastCorner"": {
+                    ""x"": 2,
+                    ""y"": 2
+                },
+                ""players"": [
+                    {
+                        ""id"": 1,
+                        ""score"": 0,
+                        ""health"": 5,
+                        ""location"": {
+                            ""x"": 2,
+                            ""y"": 2
+                        },
+                        ""orientation"": ""east""
+                    }
+                ],
+                ""pickups"": [
+                    {
+                        ""type"": ""boost"",
+                        ""location"": {
+                            ""x"": 1,
+                            ""y"": 3
+                        }
+                    }
+                ],
+                ""scoreLocations"": [
+                    {
+                        ""location"": {
+                            ""x"": 1,
+                            ""y"": 3
+                        }
+                    }
+                ],
+                ""obstacles"": [
+                    {
+                        ""location"": {
+                            ""x"": 0,
+                            ""y"": 1
+                        },
+                        ""width"": 2,
+                        ""height"": 1,
+                        ""type"": ""van"",
+                        ""orientation"": ""east""
+                    }
+                ]
+
+            }";
+
+            GameStateDTO gameState = JsonUtility.FromJson<GameStateDTO>(gameUpdateJSON);
+            Assert.AreEqual(Era.LessFlat, gameState.EraType);
+            Assert.AreEqual(new Location(-2, -2), gameState.southWestCorner);
+            Assert.AreEqual(new Location(2, 2), gameState.northEastCorner);
+            Assert.AreEqual(1, gameState.players.Length);
+            Assert.AreEqual(1, gameState.pickups.Length);
+            Assert.AreEqual(1, gameState.scoreLocations.Length);
+            Assert.AreEqual(1, gameState.obstacles.Length);
```

Comment:
`Length` is with upper case, `EraType` is with upper case, `northEastCorner` isn't. What's the convention here?